Ball of Duty Server/Persistence/Player.Domain.cs

```diff
@@ -19,8 +20,13 @@ public Character CurrentCharacter
             set
             {
                 _currentCharacter?.Unregister(Observation.ACQUISITION_OF_GOLD, this);
+                _currentCharacter?.Unregister(Observation.EXTERMINATION, this);
+                _currentCharacter?.Unregister(Observation.KILLING, this);
                 _currentCharacter = value;
+                _currentCharacter.HighScore = this.HighScore;
```

Comment:
what is the purpose of this? shouldnt Character.HighScore only be used in a single game session? why should it start at a highscore that the player has achieved in another game?